tourcms.js

```diff
@@ -1,799 +1,841 @@
+    }
+
+    startNewBooking(bookingData, channel) {
+        return this.request(TourCMS.C_START_NEW_BOOKING, channel, TourCMS.HTTP_VERB_POST, bookingData)
+    }
+
+    commitBooking(channel, bookingId) {
+        let xmlString = "<booking><booking_id>" + bookingId + "</booking_id></booking>";
+        let postData = this.parseXml(xmlString);
+        return this.request(TourCMS.C_BOOKING_COMMIT, channel, TourCMS.HTTP_VERB_POST, postData);
+    }
+
+
+    // Retrieving bookings
+
+    searchBookings(params = '', channel = 0) {
+        params = this.validateParams(params);
+        if (channel == 0)
+            return (this.request(TourCMS.P_BOOKING_SEARCH + params));
+        else
+            return (this.request(TourCMS.C_BOOKING_SEARCH + params, channel));
+    }
+
+    listBookings(channel = 0, params = '') {
+        let endpoint = TourCMS.C_BOOKINGS_LIST
+        if (channel == 0) {
+            endpoint = TourCMS.P_BOOKINGS_LIST
+        }
+        if (params) {
+            params = this.validateParams(params)
+            endpoint += params
+        }
+        return this.request(endpoint, channel)
+
+    }
+
+    showBooking(channel, bookingId, params) {
+        let fullparams = '?booking_id=' + bookingId + '&' + params
+        let endpoint = TourCMS.C_BOOKING_SHOW + fullparams
+        return this.request(endpoint, channel)
+    }
+
+    searchVoucher(voucherData = null, channel = 0) {
+
+        if (!voucherData) {
+            voucherData = '<voucher><barcode_data></barcode_data></voucher>'
+        }
+        let postData = this.parseXml(voucherData)
+
+        if (channel == 0) {
+            return (this.request(TourCMS.P_VOUCHER_SEARCH, channel, TourCMS.HTTP_VERB_POST, postData));
+        } else {
+            return (this.request(TourCMS.C_VOUCHER_SEARCH, channel, TourCMS.HTTP_VERB_POST, postData));
+        }
+    }
+
+    // Updating bookings
+
+    updateBooking(bookingData, channel) {
+        return (this.request(TourCMS.C_BOOKING_UPDATE, channel, TourCMS.HTTP_VERB_POST, bookingData));
+    }
+
+    createPayment(paymentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_PAYMENT_NEW, channel, TourCMS.HTTP_VERB_POST, paymentData));
+    }
+
+    logFailedPayment(paymentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_PAYMENT_FAIL, channel, TourCMS.HTTP_VERB_POST, paymentData));
+    }
+
+    spreedlyCreatePayment(paymentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_PAYMENT_SPREEDLY_NEW, channel, TourCMS.HTTP_VERB_POST, paymentData));
+    }
+
+    spreedlyCompletePayment(transactionId, channel) {
+        let endpoint = TourCMS.C_BOOKING_PAYMENT_SPREEDLY_COMPLETE + '?id=' + transactionId
+        return (this.request(endpoint, channel, TourCMS.HTTP_VERB_POST));
+    }
+
+    cancelBooking(channel, bookingId, note) {
+        let xmlString = "<booking><booking_id>" + bookingId + "</booking_id><note>" + note + "</note></booking>";
```

Comment:
Better to leave the user to generate the XML or being sure that we have all possible required fields

Reply:
done, let user pass XML